src/machines/java/com/enderio/machines/common/blockentity/ImpulseHopperBlockEntity.java

```diff
@@ -0,0 +1,104 @@
+package com.enderio.machines.common.blockentity;
+
+import com.enderio.machines.common.MachineTier;
+import com.enderio.machines.common.blockentity.base.PowerConsumingMachineEntity;
+import com.enderio.machines.common.init.MachineCapacitorKeys;
+import com.enderio.machines.common.io.item.MachineInventoryLayout;
+import com.enderio.machines.common.menu.ImpulseHopperMenu;
+
+import net.minecraft.core.BlockPos;
+import net.minecraft.world.entity.player.Inventory;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.inventory.AbstractContainerMenu;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.level.block.entity.BlockEntityType;
+import net.minecraft.world.level.block.state.BlockState;
+
+public class ImpulseHopperBlockEntity extends PowerConsumingMachineEntity{
+    private static final int IMPULSE_HOPPER_POWER_USE_PER_ITEM = 10; //TODO MachineCapacitorKeys.IMPULSE_HOPPER_ENERGY_CONSUME_ITEM
+
+    public ImpulseHopperBlockEntity(BlockEntityType<?> type, BlockPos worldPosition, BlockState blockState) {
+        super(MachineCapacitorKeys.IMPULSE_HOPPER_ENERGY_CAPACITY.get(),
+            MachineCapacitorKeys.IMPULSE_HOPPER_ENERGY_TRANSFER.get(),
+            MachineCapacitorKeys.IMPULSE_HOPPER_ENERGY_CONSUME.get(),
+            type, worldPosition, blockState);
+    }
+
+    @Override
+    public AbstractContainerMenu createMenu(int containerId, Inventory inventory, Player player) {
+        return new ImpulseHopperMenu(this, inventory, containerId);
+    }
+
+    @Override
+    public MachineInventoryLayout getInventoryLayout() {
+        return MachineInventoryLayout.builder(true).inputSlot(6).outputSlot(6).ghostSlot(6).build(); // first 6 input, second 6 output, third 6 ghost
+    }
+
+    @Override
+    public void serverTick() {
+        if(shouldPassItems() && ShouldActTick()) {
+            passItems();
+        }
+        super.serverTick();
+    }
+
+    public boolean ShouldActTick() {// TODO General tick method for power consuming devices?
```

Comment:
rename to follow java standards
```suggestion
    public boolean shouldActTick() {// TODO General tick method for power consuming devices?
```

Reply:
Fixed in next commit